hail/python/hail/plot/plots.py

```diff
@@ -320,100 +320,445 @@ def set_font_size(p, font_size: str = '12pt'):
+
+    # Add legend / color bar
+    legend = Legend(visible=False, click_policy='hide', orientation='vertical') if initial_col not in factor_cols else Legend(items=legend_items[initial_col], click_policy='hide', orientation='vertical')
+    color_bar = ColorBar(visible=False) if initial_col not in continuous_cols else ColorBar(color_mapper=color_mappers[initial_col])
+    sp.add_layout(legend, 'left')
+    sp.add_layout(color_bar, 'left')
+
+    return sp, legend_items, legend, color_bar, color_mappers, all_renderers
+
+
+@typecheck(x=expr_numeric, y=expr_numeric,
+           label=nullable(dictof(str, expr)), title=nullable(str),
+           xlabel=nullable(str), ylabel=nullable(str),
+           source_fields=nullable(dictof(str, expr)), colors=nullable(dictof(str, bokeh.models.mappers.ColorMapper)),
+           width=int, height=int, n_divisions=int, missing_lable=str)
+def scatter(
+        x: hl.expr.NumericExpression,
+        y: hl.expr.NumericExpression,
+        label: Dict[str, hl.expr.Expression] = None,
+        title: str = None,
+        xlabel: str = None,
+        ylabel: str = None,
+        size: int =4,
+        source_fields: Dict[str, hl.expr.Expression] = None,
+        colors: Dict[str, ColorMapper] = None,
+        width: int = 800,
+        height: int = 800,
+        n_divisions: int = None,
+        missing_label: str = 'NA'
+) -> Column:
+    """Create an interactive scatter plot.
+
+       ``x`` and ``y`` must both be a :class:`NumericExpression` from the same :class:`Table`.
+
+       This function returns a :class:`bokeh.plotting.figure.Column` containing:
+       - a :class:`bokeh.models.widgets.Select` selection widget if multiple entries are specified in the ``label``
+       - a :class:`bokeh.plotting.figure.Figure` containing the interactive scatter plot
+
+       Points will be colored by one of the labels defined in the ``label`` using the color scheme defined in
+       the corresponding entry of ``colors`` if provided (otherwise a default scheme is used). To specify your color
+       mapper, check `the bokeh documentation <https://bokeh.pydata.org/en/latest/docs/reference/colors.html>`__
+       for CategoricalMapper for categorical labels, and for LinearColorMapper and LogColorMapper
+       for continuous labels.
+       For categorical labels, clicking on one of the items in the legend will hide/show all points with the corresponding label.
+       Note that using many different labelling schemes in the same plots, particularly if those labels contain many
+       different classes could slow down the plot interactions.
+
+       Hovering on points will display their coordinates, labels and any additional fields specified in ``source_fields``.
+
+        Parameters
+        ----------
+        x : :class:`.NumericExpression`
+            List of x-values to be plotted.
+        y : :class:`.NumericExpression`
+            List of y-values to be plotted.
+        label : Dict[str, :class:`.Expression`]
```

Comment:
would it be hard to extend this to support the original interface of `label=mt.population`?

Reply:
No, that was trivial actually! I've pushed a revised version that supports this. I did not add support for `List[label]` though (this was used when providing python lists rather than hail exp.

Reply:
Actually, in the same vein, let me return a `Figure` in case no label or a single label is passed.

Reply:
Done! So this is now _mostly_ back compatible

Reply:
I could re-add the `legend` option to if desired

Reply:
nah that's OK